From c72ad8a04629ee74f32e13b2a43eb08e9fe31787 Mon Sep 17 00:00:00 2001
From: Drew Moseley <drew@moseleynet.net>
Date: Mon, 18 May 2026 16:01:35 -0400
Subject: [PATCH 4/5] arm: aquila-am69: add boot filesystem support to
 defconfig

j784s4_evm_a72_defconfig does not enable MMC commands or FAT/ext4
filesystem support in main U-Boot (only SPL).  Without these,
bootflow scan finds no bootflows and 'mmc' is an unknown command.

Add CONFIG_CMD_MMC, CMD_FAT/FS_FAT/FS_FAT_MAX_CLUSTSIZE, CMD_EXT4,
CMD_EXT4_WRITE, CMD_FS_GENERIC so the script bootmeth can access
the FAT boot partition and load boot.scr.

Add CONFIG_LEGACY_IMAGE_FORMAT so the script bootmeth's 'source'
command can parse the mkimage-wrapped boot.scr (legacy uImage format,
arch=PowerPC header set by mkimage default when -A is omitted).

Signed-off-by: Drew Moseley <drew.moseley@toradex.com>
---
 configs/aquila-am69_a72_defconfig | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/configs/aquila-am69_a72_defconfig b/configs/aquila-am69_a72_defconfig
index 3c90449a723..756a0e7e73e 100644
--- a/configs/aquila-am69_a72_defconfig
+++ b/configs/aquila-am69_a72_defconfig
@@ -22,6 +22,7 @@ CONFIG_ENV_SIZE=0x2000
 CONFIG_ENV_OFFSET=0xFFFFDE00
 CONFIG_SYS_PROMPT="Aquila AM69 # "
 CONFIG_BOOTDELAY=1
+CONFIG_LEGACY_IMAGE_FORMAT=y
 CONFIG_DISTRO_DEFAULTS=y
 CONFIG_BOOTCOMMAND="bootflow scan -b"
 CONFIG_USE_PREBOOT=y
@@ -41,3 +42,10 @@ CONFIG_FASTBOOT_FLASH=y
 CONFIG_FASTBOOT_UUU_SUPPORT=y
 CONFIG_FASTBOOT_FLASH_MMC_DEV=0
 CONFIG_FASTBOOT_CMD_OEM_FORMAT=y
+CONFIG_CMD_MMC=y
+CONFIG_CMD_FAT=y
+CONFIG_FS_FAT=y
+CONFIG_FS_FAT_MAX_CLUSTSIZE=16384
+CONFIG_CMD_EXT4=y
+CONFIG_CMD_EXT4_WRITE=y
+CONFIG_CMD_FS_GENERIC=y
-- 
2.54.0

